ExtraDry/Sample.Data/Services/RegionService.cs

```diff
@@ -48,9 +53,9 @@ public async Task<Region> RetrieveAsync(string code)
     public async Task UpdateAsync(string code, Region item)
     {
         var existing = await RetrieveAsync(code);
-        if(existing.ParentCode != item.ParentCode) {
-            var parent = await RetrieveAsync(item.ParentCode);
-            existing.SetParent(parent);
+        if(existing.Parent != null && item.Parent != null && existing.Parent.Slug != item.Parent.Slug) {
```

Comment:
Transaction boundary around `MoveSubtree` and `UpdateAsync`?

Reply:
MoveSubtree creates and commits its own transaction, as well as checking there's nothing that would complicate issues

Reply:
Still these two should be in a transaction.  What if the Update failed for some reason, even a deadlock.  Then the tree would have changed without the same in the update.  E.g. change AU-NSW to be under NZ, with the tree change occurring and the slug change not occurring.  The inner-transaction will identify that it's already in a transaction and work properly.

Reply:
Done. I added a transaction around this whole update. I've needed to have the moving of the subtree only create a transaction if there's not one existing, otherwise it tried to create a transaction within a transaction and caused other issues.